### PR TITLE
Added support for api 4.0.0-Beta-3

### PR DESCRIPTION
**Pull request by fasterfish**

### CODE REVIEW

**Comment by fasterfish** [comment]
I think the tests failed, because api-version at https://rp.epam.com lower than 4.0.

**Comment by nvborisenko** [comment]
Tests are failing because of "Unrecognized field \"retry\" (class com.epam.ta.reportportal.ws.model.StartTestItemRQ)".
I think `retry` should be passed when finihing test item.
@fasterfish please review your tests, rp.epam.com was updated to 3.2.2 which is compatible with 4.0

**Comment by fasterfish** [comment]
@nvborisenko [all tests](https://ci.appveyor.com/project/nvborisenko/client-net/build/83/tests) successfully passed.

File: ReportPortal.Client.Tests/LaunchItem/LaunchItem.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 namespace ReportPortal.Client.Tests.LaunchItem
 {
+    [Trait("LaunchItem", "")]
```
Comment (nvborisenko) [inline ReportPortal.Client.Tests/LaunchItem/LaunchItem.cs]: Incorrect usage of traits. Need smt like `Trait("Endpoint", "Launches")`. The 1st argument is category name.

File: ReportPortal.Client/Requests/FinishTestItemRequest.cs [hunk 2]
```diff
@@ -36,14 +37,17 @@ public DateTime EndTime
         public string StatusString { get { return EnumConverter.ConvertFrom(Status); } set { Status = EnumConverter.ConvertTo<Status>(value); } }
 
         public Status Status = Models.Status.Passed;
-
-        [DataMember(Name = "retry", EmitDefaultValue = false)]
```
Comment (nvborisenko) [inline ReportPortal.Client/Requests/FinishTestItemRequest.cs]: Regarding api docs FinishTestItemRequest has Retry property too. Could you please return it back?

Comment (fasterfish) [reply ReportPortal.Client/Requests/FinishTestItemRequest.cs]: done

**Pull request merged**